target/scripts/startup/setup.d/security/rspamd.sh

```diff
@@ -304,6 +305,56 @@ function __rspamd__setup_check_authenticated() {
   fi
 }
 
+# If `RSPAMD_SPAM_SUBJECT` is not empty, we create a Sieve script that alters the `Subject`
+# header, in order to prepend a user-defined string.
+function __rspamd__setup_spam_subject() {
+  if [[ -z ${RSPAMD_SPAM_SUBJECT} ]]
+  then
+    __rspamd__log 'debug' 'Spam subject is not set - no prefix will be added to spam e-mails'
+  else
+    __rspamd__log 'debug' "Spam subject is set - the prefix '${RSPAMD_SPAM_SUBJECT}' will be added to spam e-mails"
```

Comment:
This is for a major release where breaking changes could be introduced.

It should be viable to avoid introducing a separate rspamd equivalent here, and rename `SA_SPAM_SUBJECT` to `DMS_SPAM_SUBJECT` or similar.

Breaking:
- Switch to opt-in by default with explicit value. It doesn't make sense to use in IMAP, while POP3 has been proven to require other explicit config changes (_but I'm open to complicating this for POP3 to default to a fixed value_).
- ENV rename, we could just log warning or temporarily handle `SA_SPAM_SUBJECT` when it's configured explicitly.
- Handled in Dovecot instead of Postfix... the subject rewrite would only happen when delivering locally, other external MTAs mail is forwarded / relayed to would have to do their own spam subject modification on their end (_and maybe they do, but then our current DMS support would presumably prepend the subject twice then which is a bug IMO_).

---

I just don't see the point in separate logic for the two when it's effectively the same feature?

![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/ab91a4bc-a935-40cb-a2c0-c51629f5e1af)

Granted.. the `_SCORE_` feature is an extra.
- Sieve can do that similar to how the subject header value was extracted. Presumably if `spamtest` is configured for a score, you'd also have generic sieve script access to that value (_or the confidence percent representation_).
- ~~Little bit more involved to support, but looking at [this sieve language reference](https://proton.me/support/sieve-advanced-custom-filters), it should be possible to match a variable containing `*_SCORE_*` where each `*` becomes `${1}` and `${2}` we can reconstruct the spam prefix.~~ Scratch that we have bash 😆 

```bash
# Use sed to replace the `_SCORE_` token to fixed string `${spam_score}` for referencing a sieve variable:
local PREFIX_SUBJECT_IF_SPAM=sed 's/_SCORE_/${spam_score}/' <<< "${DMS_SPAM_SUBJECT}"
```

```sieve
if header :matches "X-Spam-Score" "*" {
  set "spam_score" "\${1}";
}
....
addheader :last "Subject" "${PREFIX_SUBJECT_IF_SPAM}\${subject}";
```

For rspamd, the FAQ [states that the score isn't really intended for external use](https://www.rspamd.com/doc/faq.html#why-do-i-have-zero-score-for-a-spam-message), they encourage using actions.
- [`spam-header`](https://www.rspamd.com/doc/modules/milter_headers.html#spam-header) An alternative approach to having rspamd configure a header that a sieve script may operate on.
- [`x-spamd-result`](https://www.rspamd.com/doc/modules/milter_headers.html#x-spamd-result-158) (_which our docs touch on a little bit_), the bulky series of score values, where I didn't find a clear explanation for the two values in the `default` field it contains.
- [`x-spam-level`](https://www.rspamd.com/doc/modules/milter_headers.html#x-spam-level) similar to SpamAssassin header, compatible with one of the Dovecot Sieve spamtest configs (`strlen`).

If wanting to have this `_SCORE_` feature to be anti-spam service agnostic, that may need to be explored further 🤷‍♂️

Reply:
I don't like the `_SCORE_` feature at all (because of maintainability concerns), but I agree that this should be agnostic to the choice of anti-spam software. I will rebase this PR.

Reply:
Perhaps add a deprecation notice with it then to encourage anyone that actually depends upon it to voice that concern.

Sometimes I think we carry features that only have a few users (namely the original contributor) who may no longer use DMS 🤷‍♂️ 

It doesn't seem too difficult to support, but at the same time with a sieve solution, we technically don't have to carry it anymore ourselves and can just document an example sieve script that could be added / adapted. Then we don't have to maintain any extra logic/ENV 👍

Reply:
I will add the deprecation notice in the changelog.